finalfrontier/src/dep_trainer.rs

```diff
@@ -81,15 +82,18 @@ impl<R> TrainIterFrom<Sentence> for DepembedsTrainer<R>
 where
     R: Rng,
 {
-    type Iter = Box<Iterator<Item = (usize, Vec<usize>)>>;
+    type Iter = Box<Iterator<Item = (WordWithSubwordsIdx, Vec<usize>)>>;
```

Comment:
It's the same, but can this be `Item = (Self::Focus, Vec<usize>)`? This would communicate the semantics more clearly.